language/tools/move-package/src/source_package/parsed_manifest.rs

```diff
@@ -43,7 +43,14 @@ pub struct PackageInfo {
 }
 
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub struct Dependency {
+pub enum Dependency {
+    /// Parametrised by the binary that will resolve packages for this dependency.
+    External(Symbol),
+    Internal(IntDependency),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct IntDependency {
```

Comment:
This name is confusing IMO. `InternalDependency` is much clearer and worth the extra typing :)